nncf/experimental/common/tensor_statistics/collectors.py

```diff
@@ -814,6 +825,21 @@ def _aggregate_impl(self) -> Dict[float, Tensor]:
         return retval
 
 
+class HAWQAggregator(AggregatorBase):
+    def __init__(self, num_samples: Optional[int] = None):
+        super().__init__(num_samples=num_samples)
+        self._container = Tensor(0.0)
+
+    def _register_reduced_input_impl(self, x: TensorType) -> None:
+        trace = fns.sum(fns.multiply(x, x))
+        # NOTE: average trace?? divide by number of diagonal elements
+        # TODO: revise this formula as possibly it is with an error; adopted from previous HAWQ implementation
+        self._container = (self._container + trace) / x.size
```

Comment:
@ljaljushkin, please check the formula is correct.

Reply:
I've checked. The old behavior is preserved.